### PR TITLE
Improve Event Recs by grouping PSU buildings (strip room #) and normalizing duration & attendance (TC #2 Stretch)

### PR DESCRIPTION
**Pull request by Calebc4554**
# Description  
- What does this PR do?  
  Implements smarter event recommendations by:  
  - Adding an `academicBuildings` constant and a `getLocationCategory` helper that buckets PSU buildings (strips room numbers) vs virtual/other  
  - Normalizing `duration` and `attendees` counts in `getEventFeedbackVectors`  
  - Converting `getEventFeedbackVectors` to async and adding `await` where it’s called to fix empty recommendation results  

- Why is it necessary or valuable?  
 Previously recommendations ignored location, length, and popularity metadata. These changes make reccomendations more relevant to student events on campus.  

---

# Milestones / Related Work  
- Milestone or version target: - [4.12 Use additional attributes from the event  from DB in rec system (TC #2 Stretch Feature)](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=id.ie6q2vfammsc)

---

# Test Plan  
- Manually “Like” a few events and verify they show up under “Recommended”  
- Confirm console logs show non-empty `vectors`, `clusters`, and `recIdsRaw`  
- Edge cases: events with no duration or no registrations  

---

# Additional Notes  
- Reviewer requests: check regex for room stripping

### CODE REVIEW

File: frontend/src/utils/feedbackUtils.js [hunk 1]
```diff
@@ -36,12 +37,54 @@ export async function getUserFeedbackMap() {
 	}
 }
 
+// helper to categorize locations into buckets
+function getLocationCategory(raw) {
+	const l = (raw || "").toLowerCase().trim();
```
Comment (alegleason) [inline frontend/src/utils/feedbackUtils.js:42]: Also can we have a more descriptive variable name for `l`? I assume it's location but hard to understand at a glance.

File: frontend/src/components/constants/academicBuildings.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		"wartik lab",
+		"academic activities building",
+		"outreach building",
+		"mcallister building",
+		"computer building",
+		"music building i",
+		"music building ii",
+		"sparks",
+		"boucke",
+	];
```
Comment (alegleason) [inline frontend/src/components/constants/academicBuildings.js:61]: I assume these will change with time as new buildings are added / removed / renamed. Is there maybe an API from Penn State that provides this list?
